.local/bin/searchfiles

```diff
@@ -0,0 +1,17 @@
+#!/bin/dash
+p() { printf "%s\n" "${@}"; }
+d() { dmenu -l "15" -p "${1}"; }
+f() {
+        r="$(locate -b -e --regex -d "${XDG_CONFIG_HOME}/.p.db" "(${1})$")"
```

Comment:
```suggestion
        r="$(locate -b -e --regex "(${1})$")"
```
This can also work. It uses the default `plocate` database instead of this custom one. By default, Arch Linux auto-updates the database once a day (if `plocate` is installed). Otherwise, running `sudo updatedb` will manually refresh it when needed.

By default, `plocate` scans everything, which can add a lot of junk to the output. For example, the "Image" shows the thumbnails generated in `~/.cache/lf` and `~/.cache/ueberzug`. This can be fixed by creating an `/etc/updatedb.conf` file with something like this <sup>[1](#myfootnote1)</sup>:
```
PRUNENAMES=".git .bzr .hg .svn .cache .local .cargo .librewolf"
PRUNEPATHS="/boot /dev /etc /opt /proc /run /srv /sys /tmp /usr /var"
```
This restricts scanning to only `/home`, `/root` and a few other places, leading to a cleaner file list and faster database updates.

Overall, nice work! Thanks for sharing!

<a name="myfootnote1">1</a>: [updatedb.conf man page](https://plocate.sesse.net/updatedb.conf.5.html)